src/base/memory_monitor_sr.cpp

```diff
@@ -312,10 +351,41 @@ namespace cubperf
       }
   }
 
-  int mmon_module::aggregate_stats (const MMON_MODULE_INFO &info)
+  int mmon_module::aggregate_stats (MMON_MODULE_INFO &info, bool get_brief_info) const
   {
-    int error = NO_ERROR;
-    return error;
+    strncpy (info.name, m_module_name.c_str (), m_module_name.size ());
+
+    if (get_brief_info)
+      {
+	info.stat.cur_stat = m_stat.cur_stat.load ();
+	return NO_ERROR;
```

Comment:
Code that returns in the case of if and does not return in the else is less readable.

```
if (is_brief)
{
    // gather cur_stat
}
else
{
    // gather all
    // gather comps and sub-comps
}

return return_code(error_code);
```